aten/src/ATen/core/PythonFallbackKernel.cpp

```diff
@@ -52,10 +52,9 @@ void pythonFallback(const c10::OperatorHandle& op, torch::jit::Stack* stack) {
 
 
   // If Torch Dispatch Mode is active, use its PyInterpreter for dispatch
-  const auto mode_stack_len = c10::impl::TorchDispatchModeTLS::stack_len();
-  if (mode_stack_len > 0) {
-    const auto& cur_torch_dispatch_mode_state = c10::impl::TorchDispatchModeTLS::get_stack_at(mode_stack_len - 1);
-    cur_torch_dispatch_mode_state->pyinterpreter()->dispatch(op, stack);
+  const auto maybe_mode = c10::impl::TorchDispatchModeTLS::maybe_highest_mode();
+  if (maybe_mode != c10::nullopt) {
+    (*maybe_mode)->pyinterpreter()->dispatch(op, stack);
```

Comment:
What exactly happens when we're inside a FakeTensor mode and we want to create a separate, unrelated FakeTensor mode? Is it possible to disable the current FakeTensor mode and begin another FakeTensor mode?

Reply:
I think this is still doable with a single `fake_mode_` slot. Although... thanks for the comment - I don't think this will work automatically with this PR in the current way I've implemented `FakeTensorMode.__enter__`.

I think I need something like this:

```
def __enter__(self):
    curr_mode = torch._C._maybe_get_fake_mode()
    if curr_mode is not None:
        # happy path: no existing fake mode is active
        torch._C._set_fake_mode(self)
        return
    if curr_mode is self
        # trying to re-entrantly use the same fake mode should no-op
        return
    # the case you mentioned: an ambient fake mode is active, but we want to push a new one.
    # Need to temporarily pop the current fake mode, and push it back on in `__exit__()`
    saved_mode = torch._C._unset_fake_mode()
    torch._C._set_fake_mode(self)
```